packages/backend/migrations/src/migrations/20231124095547-init-bdd.ts

```diff
@@ -0,0 +1,171 @@
+
+import type { Database } from '@app/types';
+
+export async function down(db: Kysely<Database>): Promise<void> {
+  // Migration code that reverts the database to the previous state.
+  await db.schema.dropTable('user').ifExists().execute();
+  await db.schema.dropTable('trip').ifExists().execute();
+  await db.schema.dropTable('car').ifExists().execute();
+  await db.schema.dropTable('car_type').ifExists().execute();
+  await db.schema.dropTable('reservation').ifExists().execute();
+  await db.schema.dropTable('reservation_seat').ifExists().execute();
+  await db.schema.dropTable('checkpoint_trip').ifExists().execute();
+  await db.schema.dropTable('notice').ifExists().execute();
+  await db.schema.dropTable('messaging').ifExists().execute();
+}
+
+export async function up(db: Kysely<Database>): Promise<void> {
+  // Migration code that update the database to the desired state.
+  await db.schema
+    .createTable('user')
+    .addColumn('id', 'int8', (col) =>
+      col.autoIncrement().primaryKey().unsigned(),
+    )
+    .addColumn('firstname', 'varchar(100)', (col) => col.notNull())
+    .addColumn('lastname', 'varchar(100)', (col) => col.notNull())
+    .addColumn('email', 'varchar(100)', (col) => col.notNull().unique())
+    .addColumn('password', 'varchar(255)', (col) => col.notNull())
+    .addColumn('birthdate', 'datetime', (col) => col.notNull())
+    .addColumn('biography', 'text')
+    .addColumn('avatar', 'varchar(255)')
+    .addColumn('passenger_kilometer_traveled', 'decimal(8, 1)', (col) =>
+      col.notNull().unsigned(),
+    )
+    .addColumn('driver_kilometer_traveled', 'decimal(8, 1)', (col) =>
+      col.notNull().unsigned(),
+    )
+    .addColumn('economy_achieved', 'int4', (col) => col.notNull().unsigned())
+    .addColumn('created_at', 'datetime', (col) => col.notNull())
+    .execute();
+
+  await db.schema
+    .createTable('trip')
+    .addColumn('id', 'int8', (col) =>
+      col.autoIncrement().primaryKey().unsigned(),
+    )
+    .addColumn('driver_id', 'int8', (col) =>
+      col.references('driver.id').notNull().unsigned(),
+    )
+    .addColumn('created_at', 'datetime', (col) => col.notNull())
+    .addColumn('date', 'datetime', (col) => col.notNull())
+    .addColumn('kilometer', 'decimal(6, 1)', (col) => col.notNull().unsigned())
+    .addColumn('travel_time', 'datetime', (col) => col.notNull())
+    .addColumn('seat_available', 'int2', (col) => col.notNull().unsigned())
+    .addColumn('price', 'int2', (col) => col.notNull().unsigned())
+    .addColumn('is_auto_validation', 'boolean', (col) => col.notNull())
+    .addColumn('comment', 'text')
+    .addColumn('is_baby_allowed', 'boolean', (col) => col.notNull())
+    .addColumn('is_non_vaccinated_allowed', 'boolean', (col) => col.notNull())
+    .addColumn('is_animal_allowed', 'boolean', (col) => col.notNull())
+    .addColumn('is_smoker_allowed', 'boolean', (col) => col.notNull())
+    .addColumn('is_with_toll', 'boolean', (col) => col.notNull())
```

Comment:
```suggestion
    .addColumn('has_tolls', 'boolean', (col) => col.notNull())
```
"est avec péage" -> "a (des) péages"